### PR TITLE
Bit shift operations for FixedString and String data types.

### LINKED ISSUE #27763: Bit shift operations for `FixedString` and `String` data types.
**Use case**

#8922

**Example**

```
SELECT bitShiftRight(toFixedString('abc', 3), 10)
```

### PR DESCRIPTION
**Pull request by nicelulu**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
This closes https://github.com/ClickHouse/ClickHouse/issues/27763.

- [x] Support bit shift right for FixedString and String data types.
- [x] Support bit shift left for FixedString and String data types.

### CODE REVIEW

File: src/Functions/bitShiftRight.cpp [hunk 1]
```diff
@@ -29,6 +31,51 @@ struct BitShiftRightImpl
...
+            {
+                op_pointer--;
+                out--;
+                UInt8 temp_value = *op_pointer >> shift_right_bits;
+                if (op_pointer - 1 >= begin)
+                {
+                    *out = UInt8(UInt8(*(op_pointer - 1) << (8 - shift_right_bits)) | temp_value);
+                }
+                else
+                    *out = temp_value;
```
Comment (CurtizJ) [inline src/Functions/bitShiftRight.cpp:73]: Please, add a comment, that here we combining bits from current and previous bytes.

File: src/Functions/bitShiftLeft.cpp [hunk 2]
```diff
@@ -17,6 +17,7 @@ struct BitShiftLeftImpl
 {
     using ResultType = typename NumberTraits::ResultOfBit<A, B>::Type;
     static const constexpr bool allow_fixed_string = false;
+    static const constexpr bool allow_string_integer = false;
```
[the remark this answers is not in the document]
Comment (nicelulu) [reply src/Functions/bitShiftLeft.cpp:20]: Thank you for your reviews, I want to implement String and FixedString operations on bitShiftRight first, and then on bitShiftLeft.

File: tests/queries/0_stateless/02016_bit_shift_right_for_string_integer.sql [hunk 3]
```diff
@@ -0,0 +1,44 @@
+SELECT bin('Hello') == bin(bitShiftRight('Hello', 0));
```
[the remark this answers is not in the document]
Comment (nicelulu) [reply tests/queries/0_stateless/02016_bit_shift_right_for_string_integer.sql:2]: done.

**Review by CurtizJ — APPROVED** [review]

**Comment by CurtizJ** [comment]
Stress test failure in unrelated.

**Comment by gyuton** [comment]
Internal documentation ticket: DOCSUP-14792.

**Pull request merged**